src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -116,7 +116,7 @@ type ClusterSpec struct {
 	// containers have separate resources settings and the amount of resources
 	// assigned to these containers will be required on the cluster on top of
 	// the resources defined here
-	Resources RedpandaResourceRequirements `json:"resources"`
+	Resources RedpandaResourceRequirements `json:"resources,omitempty"`
```

Comment:
this needs to receive `omitempty`, because we need to make it optional AFAIK. so we can have a CR without it (w/o a default nodepool basically, only ones listed in spec.nodePools)

Reply:
If you leave that as is you will have `Resources` of type `RedpandaResourceRequirements` which if not provided it will default to 0. To differentiate not provided value and zero value I'm used to making it a pointer to a struct.

Reply:
I think it is compatible change as any user that used or did not use resources would not see any difference.

Never the less Cluster custom resource is not defined in our product. Only cloud is using it and K8S team maintains only Redpanda CRD.